app/models/manageiq/providers/redhat/infra_manager/provision/cloning.rb

```diff
@@ -10,13 +10,29 @@ def destination_image_locked?
   end
 
   def find_destination_in_vmdb(ems_ref)
-    ManageIQ::Providers::Redhat::InfraManager::Vm.find_by(:name => dest_name, :ems_ref => ems_ref)
+    if source.template?
+      ManageIQ::Providers::Redhat::InfraManager::Vm.find_by(:name => dest_name, :ems_ref => ems_ref)
+    else
+      ManageIQ::Providers::Redhat::InfraManager::Template.find_by(:name => dest_name, :ems_ref => ems_ref)
+    end
   end
 
   def prepare_for_clone_task
     raise MiqException::MiqProvisionError, "Provision Request's Destination VM Name=[#{dest_name}] cannot be blank" if dest_name.blank?
     raise MiqException::MiqProvisionError, "A VM with name: [#{dest_name}] already exists" if source.ext_management_system.vms.where(:name => dest_name).any?
 
+    prepare_clone_options
+  end
+
+  def prepare_clone_options
+    if source.template?
+      vm_clone_options
```

Comment:
Please add some comment here. I was puzzled by this code because I see `is_templare` => `true` then do vm. I noticed similar code in line 80 where comment helped.

Reply:
agreed. I've start looking into the 'clone vm' workflow and encountered that using the `source.template?` will have to be replaced by a different method to identify the existing flow.

With the current PR, we're reusing the same workflow entity for add vm and add template. However, in 'clone vm' both source and destination are vms. I'll handle it as part of 'clone vm' PR.

I'll add comment here as well.